modules/platform/nuxeo-fsexporter/src/main/java/org/nuxeo/io/fsexporter/FSExporter.java

```diff
@@ -64,6 +69,16 @@ private void serializeStructure(CoreSession session, String fsPath, DocumentMode
             for (DocumentModel child : children) {
                 serializeStructure(session, serialized.getAbsolutePath(), child, PageProvider);
             }
+        } else if (doc.hasFacet(CollectionConstants.COLLECTION_FACET)) {
+            List<DocumentModel> collectedDocs = doc.getAdapter(Collection.class)
+                                                   .getCollectedDocumentIds()
+                                                   .stream()
+                                                   .map(IdRef::new)
+                                                   .map(session::getDocument)
+                                                   .collect(Collectors.toList());
+            for (DocumentModel collected : collectedDocs) {
+                serializeStructure(session, serialized.getAbsolutePath(), collected, PageProvider);
+            }
```

Comment:
Let's be fully functional:

    String path = serialized.getAbsolutePath();
    doc.getAdapter(Collection.class)
       .getCollectedDocumentIds()
       .stream()
       .map(IdRef::new)
       .map(session::getDocument)
       .forEach(collected -> serializeStructure(session, path, collected, PageProvider));

And remove unneeded imports.

Reply:
By using this I have to surround 

> serializeStructure(session, path, collected, PageProvider));

with a try/catch block

Reply:
Is it really better than a for loop?

Reply:
Ah indeed if `serializeStructure` can throw `IOException` then we reach the limitations of lambdas...
Ok as it is then.